### PR TITLE
feat: publish lib as Lambda Layer

### PR DESCRIPTION
**Pull request by flochaz**
## Description of your changes

* Integrate layer publication in CI/CD release process
* Publish doc on layer usage
* Add example on how to use it with CDK

### How to verify this change

Release process: https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/2826141922

A public version is already available under this ARN: `arn:aws:lambda:{REGION}:094274105915:layer:AWSLambdaPowertoolsTypeScript:1` that was published during `1.0.2` release. So can be used in any deployment. An example of its usage is available in our CDK examples and be deploy simply:

```
cd examples/cdk
npm ci
npm run cdk deploy
```

Get to your AWS Console lambda , select the Lambda called `MyLayeredFunction` and check it's code and execution.

E2E tests has been added in the layer publisher as well.

### Related issues, RFCs

#826 

### PR status

***Is this ready for review?:*** NO  
***Is it a breaking change?:*** NO

## Checklist

- [x] Check tracking is in place for our layer
- [x] Add doc in SAM README on how to set the layer
- [ ] Add doc on release process with manual update doc for version bump
- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -28,6 +28,233 @@ Core utilities such as Tracer, Logger, Metrics, and Event Handler will be availa
...
+    MyLambdaFunction:
+        Type: AWS::Serverless::Function
+        Properties:
+            ...
+            Metadata: 
+              # Manage esbuild properties
+              BuildMethod: esbuild
+              BuildProperties:
+                Minify: true
+                External:s
```
Comment (dreamorosi) [inline docs/index.md:221]: Is this `s` in `External:s` (<- this) on purpose? If yes please resolve this comment.

File: examples/cdk/package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
   "devDependencies": {
     "@types/jest": "^27.4.0",
     "@types/node": "18.0.0",
-    "aws-cdk": "^2.0.0",
+    "aws-cdk": "2.27.0",
```
Comment (dreamorosi) [inline examples/cdk/package.json:29]: Why are we pinning this specific version?

**Review by dreamorosi — APPROVED** [review]
Great work Florian, thank you fro this PR.

I've left only a couple of minor comments but it's already looking good.

File: docs/index.md (hunk 1 above)
Comment (flochaz) [reply docs/index.md:221]: oups , well catched

**Review by dreamorosi — DISMISSED** [review]

**Comment by flochaz** [comment]
Docker dependency might be an overkill, thinking about moving to a simple zip bundling:
```
mkdir nodejs && cd $_
npm init -y
npm install @aws-lambda-powertools/commons @aws-lambda-powertools/logger @aws-lambda-powertools/metrics @aws-lambda-powertools/tracer
rm package.json package-lock.json
cd ..
zip -r layer.
zip nodejs
```

**Review by dreamorosi — APPROVED** [review]

**Review by ijemmy — APPROVED** [review]

**Pull request merged**